casaos-fix-docker-api-version/run.sh

```
@@ -0,0 +1,853 @@
+  apt-cache madison docker-ce | head -n 5
+  echo ""
+
+  # Hold Docker packages to prevent auto-upgrade
+  echo "Configuring Docker packages to prevent auto-upgrade..."
+  $SUDO apt-mark unhold docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin 2>/dev/null || true
+  echo ""
+
+  # Install specific Docker version compatible with CasaOS
+  echo "Installing Docker version compatible with CasaOS (24.0.7)..."
+  
+  ARCH=$(dpkg --print-architecture)
```

Shellcheck (0.11.0)

[warning] ARCH appears unused. Verify use (or export if used externally).

(SC2034)